### PR TITLE
Replace webpack with vitejs

### PR DESCRIPTION
**Pull request by peterpeterparker**
# Changes

- drop `webpack` for `vite`
  - remove webpack dependencies
  - add various vite dependencies
  - add configuration `vite.config.ts`
  - convert webpack plugins - e.g. for canister IDs or to remove `script` - to `vite.plugin.ts`
  - move and adapt `html` file for `index.html`
  - move `showcase` html and ts file to separate directory
-  set `buffer`, `process` and `stream-browserify` as dependencies instead of dev dependencies because these are such
- polyfill `buffer` manually in `index.ts`
- set script to be loaded as `module` by canister

Remaining non-production blocker tasks in https://github.com/dfinity/internet-identity/issues/1458.

### CODE REVIEW

File: .github/workflows/canister-tests.yml [hunk 1]
```diff
@@ -621,7 +621,6 @@ jobs:
 
   clean-build:
     runs-on: ${{ matrix.os }}
-    if: ${{ github.ref == 'refs/heads/main' }}
```
Comment (nmattia) [inline .github/workflows/canister-tests.yml:624]: leftover?

File: vite.config.ts [hunk 2]
```diff
@@ -0,0 +1,94 @@
+import { extname } from "path";
+import { defineConfig, UserConfig } from "vite";
+import viteCompression from "vite-plugin-compression";
+import { injectCanisterIdPlugin, stripInjectJsScript } from "./vite.plugins";
+
+const defaultConfig = (mode?: string): Omit<UserConfig, "root"> => {
+  // Path "../../" have to be expressed relative to the "root".
+  // e.g.
+  // root = src/frontend
+  // envDiv = ../../ because .env is at the base of the project
```
Comment (nmattia) [inline vite.config.ts:10]:
```suggestion
  // envDir = ../../ because .env is at the base of the project
```

File: vite.config.ts [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+  return {
+    envDir: "../../",
```
Comment (nmattia) [inline vite.config.ts:13]: @peterpeterparker can you confirm that you use this locally?

**Review by nmattia — COMMENTED** [review]
Cool! couple questions

When I do `npm run build`, the `index.html` doesn't seem to be minified. Is that on purpose?

File: .github/workflows/canister-tests.yml (hunk 1 above)
Comment (peterpeterparker) [reply .github/workflows/canister-tests.yml]: yes. redo ✅

File: vite.config.ts (hunk 3 above)
Comment (peterpeterparker) [reply vite.config.ts]: Removed.

File: vite.plugins.ts [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+  name: "html-transform";
+  transformIndexHtml(html: string): string;
+} => ({
+  name: "html-transform",
+  transformIndexHtml(html): string {
+    return html.replace(
+      `<script type="module" crossorigin src="/index.js"></script>`,
+      ``
+    );
+  },
```
Comment (nmattia) [inline vite.plugins.ts:58]: I'm afraid this is going to break silently if vite ever changes a single character from the generated `<script...>` tag. Wdyt?

Comment (peterpeterparker) [reply vite.plugins.ts:58]: It isn't like we have much choice, don't have other idea.

We can write a test that checks there is no  script with mention of `index.js` after `npm run build`?

File: vite.plugins.ts [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+export const stripInjectJsScript = (): {
+  name: "html-transform";
+  transformIndexHtml(html: string): string;
+} => ({
+  name: "html-transform",
+  transformIndexHtml(html): string {
+    return html.replace(
+      `<script type="module" crossorigin src="/index.js"></script>`,
+      ``
+    );
```
Comment (nmattia) [inline vite.plugins.ts:57]:
```suggestion
    const match = `<script type="module" crossorigin src="/index.js"></script>`;
    if (!html.includes(match)) {
      throw new Error("Expecting script tag to replace, found none");
    }
    return html.replace(match, ``);
```

File: vite.plugins.ts (hunk 4 above)
Comment (nmattia) [reply vite.plugins.ts:58]: Yeah, added suggestion

Comment (peterpeterparker) [reply vite.plugins.ts]: Thanks!

**Review by nmattia — APPROVED** [review]

**Pull request merged**